src/app/user/settings/page.tsx

```diff
@@ -1,231 +1,337 @@
 "use client";
 
-import Card from "@/components/Card";
-import { Title } from "@/components/Title";
-import FormField from "@/components/FormField";
-import { Button } from "@/components/ui/button";
-import { Alert, AlertTitle } from "@/components/ui/alert";
-import { useState, useEffect } from "react";
-import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
 import { useAuth } from "@/app/AuthProvider";
-import { auth } from "@/lib/firebase";
+
 import {
-  updateEmail as fbUpdateEmail,
-  updatePassword as fbUpdatePassword,
-  reauthenticateWithCredential,
-  EmailAuthProvider,
-  deleteUser,
-} from "firebase/auth";
+  changeEmailWithVerification,
+  reauthWithPassword,
+  changePassword,
+  removeAccount,
+  reloadAndGetFlags,
+} from "@/lib/user";
+import { Button } from "@/components/ui/button";
+import FormField from "@/components/FormField";
+import Card from "@/components/Card";
+import { Alert, AlertTitle } from "@/components/ui/alert";
+import { Title } from "@/components/Title";
 
 type Tab = "email" | "password" | "delete";
```

Comment:
This component is too complex. It would be better to manage every tab as a separate component.